sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -22,35 +22,42 @@
 #include <windows.h>
 #endif
 
-#include <codecvt>
 #include <limits>
-#include <locale>
 #include <stdexcept>
 
 namespace Azure { namespace Storage { namespace _internal {
 
 #if defined(AZ_PLATFORM_WINDOWS)
   FileReader::FileReader(const std::string& filename)
   {
+    int sizeNeeded
+        = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), -1, nullptr, 0);
```

Comment:
I recommend passing `filename.size()` instead of `-1` so the caller gets embedded `\0` characters if they ask for them instead of silently using the prefix of `filename` that precedes any such `\0` character.

Reply:
I appended a commit to fix this.